lib/fuse_lowlevel.c

```diff
@@ -320,38 +320,46 @@ int fuse_reply_iov(fuse_req_t req, const struct iovec *iov, int count)
 	return res;
 }
 
-static size_t fuse_dirent_size(size_t namelen)
-{
-	return FUSE_DIRENT_ALIGN(FUSE_NAME_OFFSET + namelen);
-}
+#ifndef IFTODT
+# define IFTODT(mode) (((mode) & 0170000) >> 12)
+#endif
 
-static void fuse_add_dirent(struct fuse_dirent *dirent, const char *name,
-			    const struct stat *stbuf, off_t off)
+static void fuse_add_dirent(struct fuse_dirent *dirent,
+                            const char *name, const size_t namelen,
+                            const size_t entlen, const size_t entlen_padded,
+			    const struct stat *stbuf, const off_t off)
```

Comment:
I think these are way too many parameters, especially since most of them are only used once.  How about this: make the new signature:

```
static void fuse_add_dirent(struct fuse_dirent *dirent, const char *value, const size_t len,
			    const struct stat *stbuf, off_t off)
```
..and then..